src/menu.c

```diff
@@ -343,8 +344,15 @@ int gophermap(state *st, char *mapfile, int depth)
 		if (type == '#') continue;
 
 		/* Stop handling gophermap? */
-		if (type == '*') return OK;
-		if (type == '.') return QUIT;
+		if (type == '*') {
+			return_val = OK;
+			goto CLOSE_FP;
+		}
+		if (type == '.') {
+			return_val = QUIT;
```

Comment:
Unneeded since return_val is already set to QUIT.

Reply:
Can you either provide reasoning for this, or else remove the line?

Reply:
Sorry, I had missed this comment.
Done.